We seem to historically prefer pthreads instead of native threads.

--- unixODBC-2.3.14/configure.ac
+++ unixODBC-2.3.14/configure.ac
@@ -438,9 +438,9 @@ AC_DEFINE([HAVE_LIBPTH], [1], [Use the -
   else
     gotthread="no";
 
-	AC_MSG_CHECKING( if os is AIX )
+	AC_MSG_CHECKING( if $host_os os is AIX or Solaris )
 	case $host_os in
-    	"aix"*)
+    	"aix"|"solaris"*)
 			raw_threads="no";
 			AC_MSG_RESULT( yes - disable check for libthread );
     	;;
